Add initial tables (#1)
turbot/steampipe-plugin-snowflake: docs/index.md

---
organization: Turbot

> category: ["saas"]

Reviewer: is this the correct category?

icon_url: "/images/plugins/turbot/snowflake.svg"

> brand_color: "#A0E3F6"

Reviewer: Where is this brand color from?

Reviewer: `#2596be`

display_name: "Snowflake"
name: "snowflake"
description: "Steampipe plugin for querying roles, databases, and more from Snowflake."
og_description: "Query Snowflake with SQL! Open source CLI. No DB required."
og_image: "/images/plugins/turbot/snowflake-social-graphic.png"
---

# Snowflake + Steampipe

[Snowflake](https://app.snowflake.com/) enables data storage, processing, and analytic solutions that are faster, easier to use, and far more flexible than traditional offerings.

[Steampipe](https://steampipe.io) is an open source CLI to instantly query cloud APIs using SQL.

> For example: List inactive users

Reviewer: ```suggestion
For example, to list inactive users:
```

```sql
select
  name,
  email,
  disabled,
  last_success_login
from
  snowflake_user
where
  (last_success_login > now() - interval '30 days')
  and last_success_login is not null;
```

```
+-----------+------------------+----------+--------------+--------------+--------------------+
| name      | email            | disabled | default_role | has_password | has_rsa_public_key |
+-----------+------------------+----------+--------------+--------------+--------------------+
| ROHIT     | rohit@xyz.com    | false    | ACCOUNTADMIN | true         | false              |
| SUMIT     | sumit@xyz.com    | false    | PUBLIC       | true         | true               |
+-----------+------------------+----------+--------------+--------------+--------------------+
```

## Documentation

- **[Table definitions & examples →](/plugins/turbot/snowflake/tables)**

## Get started

### Install

Download and install the latest Snowflake plugin:

```bash
steampipe plugin install snowflake
```

### Credentials

The Snowflake provider supports multiple ways to authenticate:

- Password
- Keypair Authentication
- OAuth Access Token
- OAuth Refresh Token

> In all authentication methods account, user, and region is required.

Reviewer: ```suggestion
For all authentication methods, `account`, `user`, and `region` are required.
```

  # password = "~dummy@pass"

  # 2 Key Pair Authentication
  # https://docs.snowflake.com/en/user-guide/key-pair-auth.html
  # private_key_path       = "/path/to/snowflake/rsa_key.p8"
  # private_key_passphrase = "abcde"

  # OR use private key directly

  # private_key            = "-----BEGIN ENCRYPTED PRIVATE KEY-----\nMIIFHzBJ....au/BUg==\n-----END ENCRYPTED PRIVATE KEY-----"
  # private_key_passphrase = "abcde"

  # 3 OAuth Access Token
  # https://docs.snowflake.com/en/user-guide/oauth-custom.html
  # oauth_access_token = "eyJraWQiOiJLWjN....jwqt1uCG8Z94ZYZp_LK3YhQbWLkWA"

  # 4 OAuth Refresh Token
  # https://developer.okta.com/docs/guides/refresh-tokens/main/
  # oauth_client_id     = "0oa44dah4cudhAkPU5d7"
  # oauth_client_secret = "wkQYoty7kCRrBzmkqBbubxK-egaJDJ5gT1BH-4b-"
  # oauth_endpoint      = "https://xyz.abc.com/oauth2/auFGTkTZs5d7/v1/token"
  # oauth_redirect_url  = "https://xy1234.ap-south-1.aws.snowflakecomputing.com/"
  # oauth_refresh_token = "0oa44dah4cudhAkPU5d70oa44dah4cudhAkPU5d7"
}
```

## Get involved

- Open source: https://github.com/turbot/steampipe-plugin-snowflake
- Community: [Slack Channel](https://steampipe.io/community/join)

> ## Configuring Snowflake Credentials

Reviewer: Can you please bring these links up into the sections above talking about these auth methods already?
